src/pages/[chapterId].tsx

```diff
@@ -18,31 +18,37 @@ const Chapter: NextPage<ChapterProps> = ({ chapterResponse: { chapter }, versesR
 
 export const getStaticProps: GetStaticProps = async ({ params }) => {
   const chapterId = String(params.chapterId);
-
-  const [chaptersResponse, chapterResponse, versesResponse] = await Promise.all([
-    getChapters(),
+  // we need to validate the chapterId first to save calling BE since we haven't set the valid paths inside getStaticPaths to avoid pre-rendering them at build time.
+  if (!isValidChapterId(chapterId)) {
+    return {
+      notFound: true,
+    };
+  }
+  const [chapterResponse, versesResponse] = await Promise.all([
     getChapter(chapterId),
     getChapterVerses(chapterId, {
       wordFields: `verse_key, verse_id, page_number, location, ${QuranFont.QPCHafs}`,
     }),
   ]);
+  // if any of the APIs have failed due to internal server error, we will still receive a response but the body will be something like {"status":500,"error":"Internal Server Error"}.
+  if (chapterResponse.status === 500 || versesResponse.status === 500) {
+    return {
+      notFound: true,
```

Comment:
Is there an alternate approach that we can follow when the backend is not responding? 

occasionally the backend might not give us a response, and a not found might be detrimental to SEO

Reply:
As per our discussion, in-case we have an error, we will return an error UI instead of the `QuranReader` and revalidate that after 35 seconds which will trigger the regeneration of the page again.